### PR TITLE
Update cluster-policies.rst docs

### PR DESCRIPTION
**Pull request by Bowrna**
This is based on an explanation about cluster policy from @potiuk in the slack channel. Having it in docs will make this feature more understandable for new users who wants to try this feature

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
I am fine - but would love if others (native English speakers) can take a look if this is good-enough explanation :)

**Review by potiuk — APPROVED** [review]

File: docs/apache-airflow/concepts/cluster-policies.rst [hunk 1]
```diff
@@ -26,9 +26,9 @@ If you want to check or mutate DAGs or Tasks on a cluster-wide level, then a Clu
 
 There are three types of cluster policy:
 
-* ``dag_policy``: Takes a :class:`~airflow.models.dag.DAG` parameter called ``dag``. Runs at load time.
-* ``task_policy``: Takes a :class:`~airflow.models.baseoperator.BaseOperator` parameter called ``task``. Runs at load time.
-* ``task_instance_mutation_hook``: Takes a :class:`~airflow.models.taskinstance.TaskInstance` parameter called ``task_instance``. Called right before task execution.
+* ``dag_policy``: Takes a :class:`~airflow.models.dag.DAG` parameter called ``dag``. Runs at load time of the dag from DagBag.
```
Comment (o-nikolas) [inline docs/apache-airflow/concepts/cluster-policies.rst]: The rest of this doc is using the all-caps DAG nomenclature, I'd stick with that. Also, a link to the DagBag class might be helpful.
```suggestion
* ``dag_policy``: Takes a :class:`~airflow.models.dag.DAG` parameter called ``dag``. Runs at load time of the DAG from DagBag.
```

File: docs/apache-airflow/concepts/cluster-policies.rst [hunk 2]
```diff
@@ -26,9 +26,9 @@ If you want to check or mutate DAGs or Tasks on a cluster-wide level, then a Clu
...
+* ``task_policy``: Takes a :class:`~airflow.models.baseoperator.BaseOperator` parameter called ``task``. The policy gets executed when task is created during parsing of the task from DagBag at the load time. This means that the whole task definition can be altered in the task policy. It does not relate to specific task running in ``dag run``. The ``task_policy`` defined is applied to all the task_instances that will be executed in the future.
```
Comment (o-nikolas) [inline docs/apache-airflow/concepts/cluster-policies.rst]:
```suggestion
* ``task_policy``: Takes a :class:`~airflow.models.baseoperator.BaseOperator` parameter called ``task``. The policy gets executed when the task is created during parsing of the task from DagBag at load time. This means that the whole task definition can be altered in the task policy. It does not relate to a specific task running in a DagRun. The ``task_policy`` defined is applied to all the task instances that will be executed in the future.
```

File: docs/apache-airflow/concepts/cluster-policies.rst [hunk 3]
```diff
@@ -26,9 +26,9 @@ If you want to check or mutate DAGs or Tasks on a cluster-wide level, then a Clu
...
+* ``task_instance_mutation_hook``: Takes a :class:`~airflow.models.taskinstance.TaskInstance` parameter called ``task_instance``. The task_instance_mutation applies not to task but to the ``task instance`` which is the instance of task that relates to particular DagRun. It is executed in "worker" not in dag file processor just before the task_instance is executed. The policy is only applied to the currently executed run (i.e. instance) of that task.
```
Comment (o-nikolas) [inline docs/apache-airflow/concepts/cluster-policies.rst]:
```suggestion
* ``task_instance_mutation_hook``: Takes a :class:`~airflow.models.taskinstance.TaskInstance` parameter called ``task_instance``. The ``task_instance_mutation`` applies not to a task but to the instance of a task that relates to a particular DagRun. It is executed in a "worker", not in the dag file processor, just before the task instance is executed. The policy is only applied to the currently executed run (i.e. instance) of that task.
```

**Review by o-nikolas — COMMENTED** [review]
Thanks for the PR @Bowrna! I just left some grammar/style suggestions.

**Comment by Bowrna** [comment]
@o-nikolas I have committed the changes you have suggested.

**Review by o-nikolas — APPROVED** [review]

**Pull request merged**